### PR TITLE
Feat/match calendar

### PR DESCRIPTION
**Pull request by brun0lass**
### This feature covers everything related to the match calendar, delivering:

- Default match calendar
- Pagination
- By League Id
- By Year 

> Note: This feature doesn't have test coverage, but an issue will be registered for us to deal with in the future.

### CODE REVIEW

**Comment by brun0lass** [comment]
I'm already aware of my token leak, and I've already updated the leaked keys, everything that has been made public is 100% rendered useless.

File: src/api-config/urls.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        Calendar: "https://footstatsapiapp.azurewebsites.net/campeonatos/850/calendario",
+    },
+
+    espn: {
+        Calendar: "https://www.espn.com.br/futebol/time/calendario/_/id/2674/santos"
+    },
+
+    apiFootball: {
+/*
+    Vem as informações gerais das partidas, podendo colocar como parametro a season e o time, com isso dentro do JSON podemos capturar isso como importante:
```
Comment (amonigor) [inline src/api-config/urls.ts:13]: Remove this whole block of comment for the future, not needed but is a nice to have.

File: src/match-calendar/match-calendar.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    constructor(private readonly matchCalendarService: MatchCalendarService) {}
+
+    @Query(() => [EndpointResponse])
+    async matchCalendar(
+        @Args('skip', { nullable: true }) skip: number, 
+        @Args('limit', { nullable: true }) limit: number,
+        @Args('league', { nullable: true }) league: number,
+        @Args('season', { nullable: true }) season: number): Promise<EndpointResponse[]> {
+        const response = await this.matchCalendarService.getCalendar(skip, limit, league, season);
+        return response
```
Comment (amonigor) [inline src/match-calendar/match-calendar.resolver.ts:16]:
```suggestion
        return await this.matchCalenderService.getCalendar(skip, limit, league, season);
```
don't needed to create a const here.

File: src/match-calendar/match-calendar.service.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+@Injectable()
+export class MatchCalendarService {
+  async getCalendar(
+  skip: number = 0, 
+  limit: number | null, 
+  leagueId: number | null, 
+  season: number = new Date().getFullYear()): Promise<EndpointResponse[]> {
+    const request = await getRequest(url.apiFootball.Matches, hApiFootball, {
+      season: season,
+      team: '128',
```
Comment (amonigor) [inline src/match-calendar/match-calendar.service.ts:17]: create a const here.

File: src/match-calendar/match-calendar.service.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    const paginatedMatches = allMatches.slice(skip, skip + (limit ?? allMatches.length));
+
+    const response: EndpointResponse[] = paginatedMatches.map((match: Match) => {
+        return {
+            status: match.fixture.status.long,
+            league: match.league,
+            round: match.league.round,
+            stadium: match.fixture.venue,
+            date: match.fixture.timestamp,
+            homeTeam: match.teams.home,
```
Comment (amonigor) [inline src/match-calendar/match-calendar.service.ts:30]:
```suggestion
            teams: {
                home: {...},
                away: {...}
            }
```

**Review by amonigor — CHANGES_REQUESTED** [review]
Fix the comments.

**Pull request closed without merging**